pkgs/development/libraries/science/math/cudnn/generic.nix

```diff
@@ -17,44 +22,48 @@
 , url
 , hash ? null
 , sha256 ? null
-, supportedCudaVersions ? []
+, supportedCudaVersions ? [ ]
 }:
 
 assert (hash != null) || (sha256 != null);
 
 let
+  inherit (cudatoolkit) cc;
+
   majorMinorPatch = version: lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version));
   version = majorMinorPatch fullVersion;
-in stdenv.mkDerivation {
-  name = "cudatoolkit-${cudatoolkit.majorVersion}-cudnn-${version}";
+
+  useRedist = libcublas != null;
```

Comment:
That can easily be moved to the inputs instead of the `? null` trap.

Reply:
You probably mean moving `libcublas` to `buildInputs`, but the intention here is _not to include `cudatoolkit`_ (which explicitly is not redistributable) for cuda>11 where we have the `redist` packages. Thus "`useRedist`"

Reply:
I thought it actually is a good idea to move the cudatoolkit-vs-redist flag to the top (I mean, it seems at least one person got confused about the meaning of the previous flag, so I might as well make it more explicit), It now has a scary name "use ... runfile", should give people the idea that we'd like to get rid of this